modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/ClayCriteriaBuilder.es.js

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ClayCriteriaGroup from './ClayCriteriaGroup.es';
+import ClayButton from './ClayButton.es';
+
+class ClayCriteriaBuilder extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			editing: false,
+			initialCriteria: this.props.criteria
+		};
+	}
+
+	render() {
+		const {conjunctions, criteria, operators, properties} = this.props;
+
+		const {editing} = this.state;
+
+		return (
+			<div className="criteria-builder">
+				<div className="criteria-builder-toolbar">
+					<ClayButton
+						label={Liferay.Language.get('edit')}
+						onClick={this._handleToggleEdit}
```

Comment:
Ya I agree that most methods are private. It mainly seemed more like a portal pattern. It also helps distinguish the functions inside and outside of a component.

To me it seems like if anything it could unnecessary, but which part was annoying to manage?